proposals/NNNN-cxx11-base.md

```diff
@@ -0,0 +1,56 @@
+<!-- {% raw %} -->
+
+# Adopt C++ 11 Base
+
+* Proposal: [NNNN](NNNN-cxx11-base.md)
+* Author(s): [Chris Bieneman](https://github.com/llvm-beanz)
+* Sponsor: TBD
+* Status: **Under Consideration**
+* Planned Version: 202y
+
+## Introduction
+
+In DXC HLSL is a set of feature extensions on top of a subset of C++ 98. C++ 98
+is now over 20 years old and most modern C++ users have adopted newer language
+constructs. This proposal suggests taking the small step of updating HLSL 202y's
+base C++ language to C++ 2011.
+
+## Motivation
+
+C++11 is over a decade old and introduced widely adopted features, many of which
+have been frequently requested additions for HLSL.
+
+## Proposed solution
+
+Adopt a C++ 11 base language and include the following C++11 features in HLSL:
+* auto
+* decltype
+* constexpr
+* C++11 scoped enumerations
+* variadic templates
+* user-defined literals
+* [C++11 attributes](/proposals/0002-cxx-attributes.md)
+* Lambda expressions
+* Static assert
+* Range-based for loops
+

```

Comment:
The above listed set of core C++11 language features appear to be a strict subset of the [full set](https://en.cppreference.com/w/cpp/11). Is the proposal to support only this subset for now with an intent to incrementally support others as deemed necessary?

Reply:
Yes, in general I think that we should never wholesale adopt all changes in a C++ version. I think we should identify specific features we want and don't want and consciously think through how those features interact with existing HLSL language features.

Each of the features I listed here needs some specific design considerations, but IMO, they're all strong candidates for inclusion in HLSL 202y.